UserDetails/forms.py

```diff
@@ -52,12 +57,141 @@ class Meta:
+
+    def verified(self):
+        if self.membership is None:
+            return None
+        return self.membership.get_verified_state()
+
+    def get_membership_model(self, user=None, new_value=True):
+        # Check input data for correctness
+        if self.user is None and user is None:
+            raise ValueError("Field does not contain user and user was not given in method")
+        if user is not None and self.user is not None and self.user != user:
+            raise ValueError("Given user differs from field user")
+
+        if self.membership is not None:
+            return self.membership
+        if self.user is not None:
+            # If there was a user given, but the link was not found. Create a new link if allowed
+            if new_value:
+                return UserMembership(related_user=self.user, association=self.association)
+        else:
+            # user originally not given. Try to find the link
+            try:
+                return self.association.usermembership_set.get(related_user=user)
+            except ObjectDoesNotExist:
+                if new_value:
+                    return UserMembership(related_user=user, association=self.association)
+        return None
+
+
+class AssociationLinkForm(forms.Form):
+
+    def __init__(self, user, *args, **kwargs):
+        super(AssociationLinkForm, self).__init__(*args, **kwargs)
+
+        self.user = user
+
+        if user is None:
+            associations = Association.objects.filter(is_choosable=True)
+        else:
+            associations = Association.objects.filter(
+                Q(is_choosable=True) |
+                (Q(is_choosable=False) & Q(usermembership__related_user=user))) \
+                .distinct().order_by('slug')
+
+        # Get all associations and make a checkbox field
+        for association in associations:
+            field = AssociationLinkField(user, association)
+
+            self.fields[field.label] = field
+
+    def clean(self):
+        cleaned_data = super(AssociationLinkForm, self).clean()
+        has_association = False
+        for key, value in self.cleaned_data.items():
+            if value:
+                has_association = True
+
+        if not has_association:
+            raise ValidationError("At least one association needs to be chosen")
+
+        return cleaned_data
+
+    def save(self, user=None):
+        """
+        Saves the association links. Removes
+        :return:
+        """
+        if self.user is None and user is None:
+            raise ValueError("Both self.user and user are None")
+        if user is None:
+            user = self.user
+
+        for key, value in self.cleaned_data.items():
+            link = self.fields[key].get_membership_model(user, new_value=value)
+            if value:
+                if link.id is None:
+                    link.save()
+                elif link.get_verified_state() == False:
```

Comment:
I think in Python `None == False`, so if you want to explicitly check for values that are `False` and not `None`, you need:
```python
elif link.get_verified_state() is False:
```

Reply:
I checked it, None == False returns False.
Also, in this particular instance, the effect would not matter as None indicates that the object has not been verified yet, which means that verified_on is already None. Setting it to None again doesn't change anything. The only effect is that it unnecessarily executes a database entry update while not changing any data.

Reply:
Okay! I saw indeed that it in this case only leads to unnecessary db executions. Interesting that `None!=False` in Python, since `bool(None)==False`, thus it apparently does some other type conversion.